### DISCUSSION TITLE
Self-referenced edge as an arrow around the node

### DISCUSSION THREAD

**Discussion by andreisemianiuk** [xyflow/xyflow#2720] [Q&A]
### Please describe the feature that you want to propose

at the moment if i want to add an edge connection for the same node( i.e. the connection starts and ends in the same node) it displays this connection as a plain line. how to make this connection go around the node for better visual perception?
![image](https://user-images.githubusercontent.com/62586934/209786264-bb09d16d-6051-44e1-914e-55e0ae1b8de6.png)

**Comment by bcakmakoglu** [comment]
Moritz did an example for this sort of case once.
[Here's the sandbox.](https://codesandbox.io/s/sharp-orla-vyjqen?file=/src/CustomEdge.tsx)

**Reply by reorx** [reply]
This sample works great, except that it doesn't show a label. Is there any hint on how to implement a label in this example that looks the same as the built-in edge types?

**Comment by reorx** [comment]
Nevermind, here's my implementation of a custom edge with the same label as builtin edges:

```ts
// ref: https://reactflow.dev/examples/edges/custom-edges
export default function SelfRefEdge(props: EdgeProps) {
  // we are using the default bezier edge when source and target ids are different
  if (props.source !== props.target) {
    return <BezierEdge {...props} />;
  }

  const { sourceX, sourceY, targetX, targetY, id, markerEnd, ...otherProps } = props;
  const radiusX = (sourceX - targetX) * 0.6;
  const radiusY = 50;
  const edgePath = `M ${sourceX - 5} ${sourceY} A ${radiusX} ${radiusY} 0 1 0 ${
    targetX + 2
  } ${targetY}`;

  return <>
    <BaseEdge path={edgePath} markerEnd={markerEnd} {...otherProps} />
    <EdgeText
      x={((sourceX + targetX) / 2)}
      y={((sourceY + targetY) / 2) - 80}
      {...otherProps}
    />
  </>
}
```
